code/modules/explorer_drone/adventure.dm

```diff
@@ -41,25 +42,137 @@
 #define REQ_VALUE_FIELD "value"
 #define REQ_OPERATOR_FIELD "operator"
 
-GLOBAL_LIST_EMPTY(explorer_drone_adventures)
+#define CURRENT_ADVENTURE_VERSION 1
 
+/// All possible adventures in raw form
+GLOBAL_LIST_EMPTY(explorer_drone_adventure_db_entries)
 
+/// Loads all adventures from DB
 /proc/load_adventures()
 	. = list()
-	for(var/filename in flist(ADVENTURE_DIR))
-		var/datum/adventure/adventure = try_loading_adventure(filename)
-		if(adventure)
-			. += adventure
+	if(!SSdbcore.Connect())
+		GLOB.explorer_drone_adventure_db_entries = .
+		return
+	var/datum/db_query/Query = SSdbcore.NewQuery("SELECT id,adventure_data,uploader,timestamp,approved FROM [format_table_name("text_adventures")]")
+	if(!Query.Execute())
+		qdel(Query)
+		return
+	while(Query.NextRow())
+		var/datum/adventure_db_entry/entry = new()
+		entry.id = Query.item[1]
+		entry.raw_json = Query.item[2]
+		entry.uploader = Query.item[3]
+		entry.timestamp = Query.item[4]
+		entry.approved = Query.item[5]
+		entry.extract_metadata()
+		. += entry
+	qdel(Query)
+	GLOB.explorer_drone_adventure_db_entries = .
+
+/datum/adventure_db_entry
+	/// db id or null for freshly created adventures
+	var/id
+	/// actual adventure json string
+	var/raw_json
+	/// ckey of last change user.
+	var/uploader
+	/// Time of last change.
+	var/timestamp
+	/// Unapproved adventures won't be used for exploration sites.
+	var/approved = FALSE
+	/// Was the adventure used for exploration site this round.
+	var/placed = FALSE
+
+	//Variables below are extracted from the JSON
+
+	/// json version
+	var/version
+	/// adventure name
+	var/name
+	/// required site traits to use this adventure
+	var/list/required_site_traits
+
+/// Check if the adventure usable for given exploration site traits
+/datum/adventure_db_entry/proc/valid_for_use(list/site_traits)
+	if(!raw_json || version != CURRENT_ADVENTURE_VERSION || placed)
+		return FALSE
+	if(required_site_traits && length(required_site_traits - site_traits) != 0)
+		return FALSE
+	return TRUE
+
+/// Creates DB entry if it does not exist in the DB yet. Otherwise updates local data if our timestamp is older or updates db data if our timestamp is newer.
```

Comment:
This isn't perfect and conflicts leading to overwritten data is still going to happen, but i guess its better then nothing.

Updating the db with changes immediately and only updating the parts that changed is how you avoid that kind of issue.

Reply:
Sure, i can change it to doing this db side, will need to split update and save on ui